tests/components/smartthings/test_config_flow.py

```diff
@@ -4,207 +4,266 @@
     flow.hass = hass
+
+    result = await flow.async_step_pat()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "pat"
+    assert result["errors"] == {}
+    assert result["data_schema"]({CONF_ACCESS_TOKEN: ""}) == {CONF_ACCESS_TOKEN: ""}
+    assert "token_url" in result["description_placeholders"]
+    assert "component_url" in result["description_placeholders"]
+
+
+async def test_step_pat_defaults_token(hass):
+    """Test pat form defaults the token from another entry."""
     token = str(uuid4())
-    entry = MockConfigEntry(domain=DOMAIN, data={"access_token": token})
+    entry = MockConfigEntry(domain=DOMAIN, data={CONF_ACCESS_TOKEN: token})
     entry.add_to_hass(hass)
+    flow = SmartThingsFlowHandler()
+    flow.hass = hass
 
-    result = await flow.async_step_user({"access_token": token})
+    result = await flow.async_step_pat()
 
+    assert flow.access_token == token
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"access_token": "token_already_setup"}
+    assert result["step_id"] == "pat"
+    assert result["errors"] == {}
+    assert result["data_schema"]({}) == {CONF_ACCESS_TOKEN: token}
+    assert "token_url" in result["description_placeholders"]
+    assert "component_url" in result["description_placeholders"]
 
 
-async def test_token_unauthorized(hass, smartthings_mock):
-    """Test an error is shown when the token is not authorized."""
+async def test_step_pat_invalid_token(hass):
+    """Test an error is shown for invalid token formats."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
+    token = "123456789"
+
+    result = await flow.async_step_pat({CONF_ACCESS_TOKEN: token})
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "pat"
+    assert result["data_schema"]({}) == {CONF_ACCESS_TOKEN: token}
+    assert result["errors"] == {"access_token": "token_invalid_format"}
+    assert "token_url" in result["description_placeholders"]
+    assert "component_url" in result["description_placeholders"]
 
+
+async def test_step_pat_unauthorized(hass, smartthings_mock):
+    """Test an error is shown when the token is not authorized."""
+    flow = SmartThingsFlowHandler()
+    flow.hass = hass
     request_info = Mock(real_url="http://example.com")
     smartthings_mock.apps.side_effect = ClientResponseError(
         request_info=request_info, history=None, status=401
     )
+    token = str(uuid4())
 
-    result = await flow.async_step_user({"access_token": str(uuid4())})
+    result = await flow.async_step_pat({CONF_ACCESS_TOKEN: token})
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"access_token": "token_unauthorized"}
+    assert result["step_id"] == "pat"
+    assert result["errors"] == {CONF_ACCESS_TOKEN: "token_unauthorized"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)